guardian/typerighter
# Bump CDK dependencies

## What does this change?

This PR bumps some CDK dependencies in order to resolve a vulnerability in `@guardian/cdk` (derived from `@babel/traverse`). None of the breaking changes in that library from `v50` or `v51` should affect Typerighter.

It also bumps the node version to`16.14.0` - as the previous version was incompatible with those dependencies. 

I tried bumping higher to `18` and `20` and strangely found that one of our Scala tests failed much more regularly (locally and in CI) when I did (`it should "reject work that exceeds its buffer size"...` in `MatcherPoolTest`). Might require some digging when we want to bump Node further.

## How to test

- Run the manager and checker locally according to the instructions in the readme to check that they are compatible with the new (very similar) Node version. Do they build and run as expected? 
- Deploy this branch to CODE. If it works, the build and deploy have worked as intended and we can be happy with the CDK changes.

## How can we measure success?

1. Package builds and deploys as expected.
2. The vulnerability is no longer reported by Snyk after merge.
